fendermint/app/src/cmd/run.rs

```diff
@@ -47,8 +56,19 @@ async fn run(settings: Settings) -> anyhow::Result<()> {
         NamespaceBlockstore::new(db.clone(), ns.state_store).context("error creating state DB")?;
 
     let resolve_pool = CheckpointPool::new();
-    let parent_finality = Arc::new(create_parent_finality(&settings).await?);
-    let app: App<_, _, AppStore, _> = App::new(
+
+    let parent_finality_provider = if settings.ipc.is_topdown_enabled() {
+        info!("topdown finality enabled");
+        let config = settings.ipc.top_down_config()?.clone();
+        Arc::new(MaybeDisabledProvider::enabled(
+            InMemoryFinalityProvider::uninitialized(config),
+        ))
+    } else {
+        info!("topdown finality disabled");
+        Arc::new(MaybeDisabledProvider::disabled())
+    };
```

Comment:
This seems very similar to the [Toggle](https://docs.rs/libp2p/latest/libp2p/swarm/behaviour/toggle/struct.Toggle.html) behaviour in libp2p, maybe you can use that name too.